swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/agents/ScheduledAgent.kt

```diff
@@ -168,21 +159,14 @@ abstract class ScheduledAgent(
 
   companion object {
     fun timeToWork(schedule: Schedule, clock: Clock): Boolean {
-      if (!schedule.enabled) {
-        return true
-      }
-
-      val startTime: LocalTime = schedule.getResolvedStartTime()
-      val endTime: LocalTime = schedule.getResolvedEndTime()
-      val now = LocalTime.from(clock.instant().atZone(schedule.getZoneId()))
-
-      return if (startTime <= endTime) {
-        // Work during the day
-        now.isAfter(startTime) && now.isBefore(endTime)
-      } else {
-        // Work during the night
-        now.isAfter(startTime) || now.isBefore(endTime)
-      }
+      return if (schedule.enabled && LocalDate.now(clock).dayOfWeek in schedule.allowedDaysOfWeek) {
+        val startTime: LocalTime = schedule.getResolvedStartTime()
+        val endTime: LocalTime = schedule.getResolvedEndTime()
+        val now = LocalTime.from(clock.instant().atZone(schedule.getZoneId()))
+
+        startTime.isBefore(endTime) &&
```

Comment:
If it's no longer valid to have a start time after your end time (ex. working from 8 pm to 8 am) then we should throw an exception or print some error message